tests/components/fitbit/test_config_flow.py

```diff
@@ -0,0 +1,266 @@
+    assert resp.status == 200
+    assert resp.headers["content-type"] == "text/html; charset=utf-8"
+
+    aioclient_mock.post(
+        OAUTH2_TOKEN,
+        json=SERVER_ACCESS_TOKEN,
+    )
+
+    requests_mock.register_uri(
+        "GET", PROFILE_API_URL, status_code=HTTPStatus.INTERNAL_SERVER_ERROR
+    )
+
+    result = await hass.config_entries.flow.async_configure(result["flow_id"])
+    assert result.get("type") == FlowResultType.ABORT
+    assert result.get("reason") == "cannot_connect"
+
+
+async def test_import_fitbit_config(
+    hass: HomeAssistant,
+    fitbit_config_setup: None,
+    sensor_platform_setup: Callable[[], Awaitable[bool]],
+    issue_registry: ir.IssueRegistry,
+) -> None:
+    """Test that platform configuration is imported successfully."""
+
+    with patch(
+        "homeassistant.components.fitbit.async_setup_entry", return_value=True
+    ) as mock_setup:
+        await sensor_platform_setup()
+
+    assert len(mock_setup.mock_calls) == 1
+    entries = hass.config_entries.async_entries(DOMAIN)
+    assert len(entries) == 1
+
+    # Verify valid profile can be fetched from the API
+    config_entry = entries[0]
+    assert config_entry.title == "My name"
+    assert config_entry.unique_id == PROFILE_USER_ID
+
+    data = dict(config_entry.data)
+    assert "token" in data
+    del data["token"]["expires_at"]
+    # Verify imported values from fitbit.conf and configuration.yaml
+    assert dict(config_entry.data) == {
+        "auth_implementation": DOMAIN,
+        "clock_format": "24H",
+        "monitored_resources": ["activities/steps"],
+        "token": {
+            "access_token": FAKE_ACCESS_TOKEN,
+            "refresh_token": FAKE_REFRESH_TOKEN,
+        },
+        "unit_system": "default",
+    }
+
+    # Verify an issue is raised for deprecated configuration.yaml
+    issue = issue_registry.issues.get((DOMAIN, "deprecated_yaml"))
+    assert issue
+    assert issue.translation_key == "deprecated_yaml_import"
+
+
+async def test_import_fitbit_config_failure_cannot_connect(
+    hass: HomeAssistant,
+    fitbit_config_setup: None,
+    sensor_platform_setup: Callable[[], Awaitable[bool]],
+    issue_registry: ir.IssueRegistry,
+    requests_mock: Mocker,
+) -> None:
+    """Test that platform configuration is imported successfully."""
+
+    requests_mock.register_uri(
+        "GET", PROFILE_API_URL, status_code=HTTPStatus.INTERNAL_SERVER_ERROR
+    )
+
+    with patch(
+        "homeassistant.components.fitbit.async_setup_entry", return_value=True
+    ) as mock_setup:
+        await sensor_platform_setup()
+
+    assert len(mock_setup.mock_calls) == 0
+
+    # Verify an issue is raised that we were unable to import configuration
+    issue = issue_registry.issues.get((DOMAIN, "deprecated_yaml"))
+    assert issue
+    assert issue.translation_key == "deprecated_yaml_import_issue_cannot_connect"
+
+
+async def test_import_fitbit_config_already_exists(
```

Comment:
We should also test a user flow with the same user already configured.

Reply:
Done.